plugin/src/main/kotlin/com/automattic/android/PublishToS3Plugin.kt

```diff
@@ -1,37 +1,48 @@
 package com.automattic.android.publish
 
+import com.android.build.gradle.LibraryExtension
 import org.gradle.api.Project
 import org.gradle.api.Plugin
 import org.gradle.api.publish.maven.tasks.GenerateMavenPom
 import org.gradle.api.publish.maven.tasks.PublishToMavenRepository
-import org.gradle.jvm.tasks.Jar
-import com.android.build.gradle.LibraryExtension
 
 class PublishToS3Plugin : Plugin<Project> {
     override fun apply(project: Project) {
-        project.plugins.apply("maven-publish")
-        project.addS3MavenRepository()
+        applyInternal(project)
```

Comment:
Minor (🔍): Consider being explicit about the named arguments and even avoid giving a default to `withSources` and the `applyInternal(...)` private function, just to make it even more explicitly and harder to mess with:

```kotlin
class PublishToS3Plugin : Plugin<Project> {
    override fun apply(project: Project) {
        applyInternal(
            project = project,
            withSources = true,
        )
    }
}

class PublishToS3WithoutSourcesPlugin : Plugin<Project> {
    override fun apply(project: Project) {
        applyInternal(
            project = project,
            withSources = false,
        )
    }
}

private fun applyInternal(
    project: Project,
    withSources: Boolean,
) {
    ...
}
```

PS: Feel free to ignore.

Reply:
I think this is more of a personal taste, but I don't feel strongly about it one way or another, so I applied your suggestion in 1882483472294e2c2eb71d76efe09e18719d06c5.

I did omit the `project = project` part though because I don't think it provides any extra context.

Reply:
Thanks @oguzkocer and I agree that this can be considered as a personal taste, thus I marked that as minor (🔍) and ask you to ignore it if you felt like it, you can ignore any minor (🔍) comment from me.

> I did omit the project = project part though because I don't think it provides any extra context.

Just an FYI that I suggested to have named arguments on both parameters so as to have a consistent Kotlin-like build pattern of sorts. It is usually recommended to either have everything as named argument or none since will make it easier to read the function's usage without entering into it.